### ISSUE TITLE
Spaces in JS block incorrectly flagged as indentation error

### ISSUE THREAD

**Issue by paulschreiber** [WordPress-Coding-Standards/WordPress-Coding-Standards#330]
Sample code:
```php
<?php
class Foo {
	public static function bar( $atts ) {
		$script = "
      <script>
        DV.load('https://www.documentcloud.org/documents/$id.js', {
          width: '" . esc_js( $width ) . "',
          height: '" . esc_js( $height ) . "',
          foo: 'bar'
        });
      </script>
      <style>#DV-viewer-{$id}</style>";

		return $script;
	}
}
```

When I run PHPCS:
```
$ phpcs -p -s -v --standard=WordPress-VIP sample.php
```

I get this output:
```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 8 | ERROR | [x] Tabs must be used to indent lines; spaces are not
   |       |     allowed
   |       |     (Generic.WhiteSpace.DisallowSpaceIndent.SpacesUsed)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

If you delete either the width or the height line, the error goes away.

**Comment by JDGrimes** [comment]
I think this is an upstream issue, since the sniff is part of PHPCS's `Generic` standard, not `WordPress`.

**Comment by paulschreiber** [comment]
I thought so too, but I couldn't reproduce this with the non-WP standards I tried.

**Comment by GaryJones** [comment]
JavaScript is supposed to be [indented with tabs](https://make.wordpress.org/core/handbook/coding-standards/javascript/#spacing) as well, for WordPress.

**Comment by westonruter** [comment]
BTW, also probably better to use `wp_json_encode()` instead of `esc_js()`:

```php
<?php
class Foo {
	public static function bar( $attrs ) {
		// ...
		$url = "";
		$attrs = array(
			'width' => $width,
			'height' => $height,
			'foo' => $bar,
		);
		$script = sprintf( 
			'<script>DIV.load( %s, %s )</script>', 
			wp_json_encode( $url ) ,
			wp_json_encode( $attrs )
		);
		// ...
		return $script;
	}
}
```

**Comment by paulschreiber** [comment]
What @GaryJones and @westonruter said is true. Regardless, there's something odd with the parser if it flags the indentation like it does — I would expect it to flag all of the lines as needing tabs (or none of them), not just one.

Should I report this to PHPCS/squiz labs?

**Comment by paulschreiber** [comment]
@westonruter Is `wp_json_encode()` a drop-in replacement for `json_encode()`? If so, should WPCS flag  uses of `json_encode()`?

**Comment by westonruter** [comment]
> Is `wp_json_encode()` a drop-in replacement for `json_encode()``? If so, should WPCS flag uses of `json_encode()`?

Good call. Yes, it is a drop-in replacement/wrapper. More info: http://pento.net/2014/12/19/json-encoding-in-wordpress-4-1/

> Regardless, there's something odd with the parser if it flags the indentation like it does — I would expect it to flag all of the lines as needing tabs (or none of them), not just one. Should I report this to PHPCS/squiz labs?

Yeah, there's probably some oddities around the parser where it switches from tabs, to a non-PHP block, then with spaces for indentation.
